businessCentral/app/src/Execution.Codeunit.al

```diff
@@ -207,6 +207,9 @@ codeunit 82569 "ADLSE Execution"
         ADLSEDeletedRecord: Record "ADLSE Deleted Record";
         DeletedTablesNottoSync: Record "Deleted Tables Not to Sync";
     begin
+        if RecRef.Number = Database::"ADLSE Deleted Record" then
```

Comment:
You cannot add the Deleted Record record in the table setup (procedure AddTableToExport in ADLSE Setup codeunit). So in that case this code is not needed.

Reply:
True but that means if you have a big deleted records table, and you try to clean it up for every delete 3 extra queries are executed. This can be bypassed using the above code. So it's not really a bug but an optimization for the deletion performance.